lib/view/weekView.dart

```diff
@@ -2,34 +2,148 @@ import 'package:flutter/material.dart';
+                        inactiveTrackColor: Colors.grey.shade400,
+                        splashRadius: 50.0,
+                        value: forEvents,
+                        onChanged: (value) {
+                          setState(() {
+                            forEvents = value;
+                          });
+                          if (!forEvents) {
+                            Navigator.of(context).push(
+                              MaterialPageRoute(
+                                builder: (context) => taskView(),
+                              ),
+                            );
+                          }
+                        },
+                      ),
+                      const Text(
+                        ' Events',
+                        style: TextStyle(
+                          color: Colors.black,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+            actions: [
+              MenuAnchor(
+                builder: (BuildContext context, MenuController controller,
+                    Widget? child) {
+                  return IconButton(
+                    color: Colors.black,
+                    onPressed: () {
+                      if (controller.isOpen) {
+                        controller.close();
+                      } else {
+                        controller.open();
+                      }
+                    },
+                    icon: const Icon(Icons.more_vert),
+                    tooltip: 'Show menu',
+                  );
+                },
+                menuChildren: [
+                  IconButton(
+                      icon: const Icon(
+                          color: Colors.black, Icons.calendar_month_rounded),
+                      onPressed: () async {
+                        DateTime? picked = await showDatePicker(
+                          context: context,
+                          initialDate: startDate,
+                          firstDate: DateTime(2000),
+                          lastDate: DateTime(2101),
+                        );
+                        if (picked != null) {
+                          setState(() {
+                            startDate = mostRecentMonday(picked);
+                          });
+                        }
+                      }),
+                  IconButton(
+                      icon: const Icon(color: Colors.black, Icons.east),
+                      onPressed: () {
+                        setState(() {
+                          startDate = startDate.add(Duration(days: 7));
+                        });
+                      }),
+                  IconButton(
+                      icon: const Icon(color: Colors.black, Icons.west),
+                      onPressed: () {
+                        setState(() {
+                          startDate = startDate.subtract(Duration(days: 7));
```

Comment:
.add and .subtract dont work well with dates when theres stuff like daylight savings
Can you switch to getDateOnly(offset: 7 or -7)?